Sources/Amplitude/Utilities/Diagonostics.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  Diagonostics.swift
+//  Amplitude-Swift
+//
+//  Created by Qingzhuo Zhen on 3/4/24.
+//
+
+import Foundation
+
+public class Diagnostics {
+    private var malformedEvents: [String]?
+    private var errorLogs: [String]?
+
+    init(){}
+
+    func addMalformedEvent(_ event: String) {
+        if malformedEvents == nil {
+            malformedEvents = [String]()
+        }
+        malformedEvents?.append(event)
+    }
+
+    func addErrorLog(_ log: String) {
+        if errorLogs == nil {
+            errorLogs = [String]()
+        }
+        errorLogs?.append(log)
+    }
+
+    func hasDiagnostics() -> Bool {
+        return (malformedEvents != nil && malformedEvents!.count > 0) || (errorLogs != nil && errorLogs!.count > 0)
+    }
+
+    /**
+     * Extracts the diagnostics as a JSON string.
+     * Warning: This will clear stored diagnostics.
+     * @return JSON string of diagnostics or empty if no diagnostics are present.
+     */
+     func extractDiagonosticsToString() -> String {
+        if !hasDiagnostics() {
+            return ""
+        }
+        var diagnostics = [String: [String]]()
+        if malformedEvents != nil && malformedEvents!.count > 0 {
+            diagnostics["malformed_events"] = malformedEvents
+        }
+        if errorLogs != nil && errorLogs!.count > 0 {
```

Comment:
NIT: Worried that the error logs may become too much for certain devices? Does it make sense to just send the last N logs?

Reply:
The error logs is more in memory now and wouldn't be included in retries now, so I guess the chance to have lots of error logs accumulating and sending is low. We could also only take first N if that is more preferred